### PR TITLE
Generate query variations based on different personas.

### PR DESCRIPTION
**Pull request by keyurva**
* Also includes script to call the detect endpoint and compare results based on number of matched attributes.
* Follow-ups: translate variations, generate comparison result csvs, docs.

### CODE REVIEW

File: tools/nl/translate_compare/detect.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    self.svs = set(context.get("svs", []))
+    self.places = set()
+    for place in context.get("places", []):
+      dcid = place.get("dcid", "")
+      if dcid:
+        self.places.add(dcid)
+
+  def compare(self, other: "Detection") -> int:
+    return len(self.places.intersection(other.places)) + len(
+        self.svs.intersection(other.svs))
```
Comment (pradh) [inline tools/nl/translate_compare/detect.py:39]: Might track them separately.

File: tools/nl/translate_compare/detect.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  def __len__(self):
+    return len(self.places) + len(self.svs)
+
+  def __str__(self) -> str:
+    return f"Attributes: {len(self)}\nPlaces: {self.places}\nSVs: {self.svs}"
+
+
+def detect(query):
+  try:
+    resp = requests.post(f"{_API_BASE_URL}/api/explore/detect?q={query}",
```
Comment (pradh) [inline tools/nl/translate_compare/detect.py:50]: add `detector=heuristic`

File: tools/nl/translate_compare/detect.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  queries = [
+      "Tell me about economic equity in California",
+      "Why is it that some people in California have lots of money and some people don't have any money?"
+  ]
+
+  result1 = detect(queries[0])
+  result2 = detect(queries[1])
+  print(result1)
+  print(result2)
+  print(result1.compare(result2))
```
Comment (pradh) [inline tools/nl/translate_compare/detect.py:72]: TODO: clean me up

**Review by pradh — APPROVED** [review]

File: tools/nl/translate_compare/detect.py (hunk 1 above)
Comment (keyurva) [reply tools/nl/translate_compare/detect.py]: Done.

File: tools/nl/translate_compare/detect.py (hunk 2 above)
Comment (keyurva) [reply tools/nl/translate_compare/detect.py]: Done.

File: tools/nl/translate_compare/detect.py (hunk 3 above)
Comment (keyurva) [reply tools/nl/translate_compare/detect.py:76]: Done.

**Pull request merged**